### PR TITLE
make set_decomposition work on new device

### PR DESCRIPTION
**Pull request by timmysilv**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/transforms/tape_expand.py [hunk 1]
```diff
@@ -350,6 +353,33 @@ def custom_decomp_expand(self, circuit, max_expansion=decomp_depth):
     return custom_decomp_expand
 
 
+@transform
+def decomp_transform(tape: QuantumTape, custom_decomps: dict) -> (Sequence[QuantumTape], Callable):
```
Comment (trbromley) [inline pennylane/transforms/tape_expand.py:357]: Any worries about making this user facing? With a bit of polish, we can have this transform version available and think about deprecation of  `set_decomposition` in 0.34.

Comment (timmysilv) [reply pennylane/transforms/tape_expand.py:357]: I also think _polish, publicize, deprecate_ makes sense.

Comment (trbromley) [reply pennylane/transforms/tape_expand.py:357]: How about `custom_decomposition` as the name?

**Comment by timmysilv** [comment]
yep.

**Pull request closed without merging**